### PR TITLE
Add ensure issue authorization action

### PR DESCRIPTION
**Pull request by kntsoriano**
https://github.com/exercism/wip/issues/36, https://github.com/exercism/wip/issues/34

Once we've merged this in, I will send a PR to exercism/wip to implement this.

### CODE REVIEW

File: ensure-issue-authorization/action.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+author: Karlo Nicholas Soriano
+inputs:
+  close-message:
+    description: Message to comment on the issue before closing
+  lock-message:
+    description: Message to comment on the issue before locking
+  lock-on-permission:
+    description: When user's permission level matches this, issue will be locked
+  close-on-permission:
+    description: When user's permission level matches this, issue will be closed
```
Comment (iHiD) [inline ensure-issue-authorization/action.yml]: What permission will you use for this? We're triggering on the lack of a write permission, so I don't know how this works? Does everyone get "read" by default?

File: ensure-issue-authorization/action.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Ensure issue authorization
+description: Ensures issues are authorized and automatically closes unauthorized issues
+author: Karlo Nicholas Soriano
+inputs:
+  close-message:
+    description: Message to comment on the issue before closing
+  lock-message:
+    description: Message to comment on the issue before locking
+  lock-on-permission:
+    description: When user's permission level matches this, issue will be locked
```
Comment (iHiD) [inline ensure-issue-authorization/action.yml]: Do we need multiple? e.g. we want to lock if someone has write or admin role. Or do both roles have the write permission? No idea how it works :)

File: ensure-issue-authorization/action.yml (hunk 1 above)
Comment (kntsoriano) [reply ensure-issue-authorization/action.yml]: Everyone gets `read` by default according to this: https://help.github.com/en/articles/repository-permission-levels-for-an-organization#permission-levels-for-repositories-owned-by-an-organization.

File: ensure-issue-authorization/action.yml (hunk 2 above)
Comment (kntsoriano) [reply ensure-issue-authorization/action.yml]: To support that, we would need to change the parameter to receive an array of roles.

I get the user's permission level from https://developer.github.com/v3/repos/collaborators/#review-a-users-permission-level and I think they're mutually exclusive.

**Comment by kntsoriano** [comment]
I'll be merging this in as it's a good first pass :) We can improve on role handling later.

**Pull request merged**